### PR TITLE
[Merged by Bors] - ET-3506 load smbert into app state (2)

### PR DESCRIPTION
**Pull request by rustonaut**
Adds CoiConfig to config and CoiSystem, Embedder to application state.

**References:**

- [ET-3506]
- based on #659 

[ET-3506]: https://xainag.atlassian.net/browse/ET-3506?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: discovery_engine_core/web-api2/src/embedding.rs [hunk 1]
```diff
@@ -42,3 +45,21 @@ impl Default for Config {
         }
     }
 }
+
+pub(crate) struct Embedder {
```
Comment (rustonaut) [inline discovery_engine_core/web-api2/src/embedding.rs:49]: :grey_exclamation:  Using a generic name allows to later one add alternative algorithms without needing to rename things everywhere. E.g. when using a different algorithm then bert for some customer using non-lating language.

File: discovery_engine_core/web-api2/src/ingestion.rs [hunk 2]
```diff
@@ -28,31 +29,43 @@ use crate::{
 pub struct Ingestion;
 
 impl Application for Ingestion {
+    type AppStateExtension = AppStateExtension;
     type ConfigExtension = ConfigExtension;
 
-    fn configure(config: &mut ServiceConfig) {
+    fn configure_service(config: &mut ServiceConfig) {
```
Comment (rustonaut) [inline discovery_engine_core/web-api2/src/ingestion.rs:35]: :grey_exclamation: Renamed to be a bit more clear about what is configured.

**Comment by rustonaut** [comment]
@Robert-Steiner ?

File: discovery_engine_core/web-api2/src/server.rs [hunk 3]
```diff
@@ -26,38 +28,33 @@ use actix_web::{
...
-    /// the config and environment.
-    #[arg(short, long)]
-    #[serde(skip_serializing_if = "Option::is_none")]
-    bind_to: Option<SocketAddr>,
-
-    /// File to log to additionally to logging to stdout.
-    #[arg(short, long)]
-    #[serde(skip_serializing_if = "Option::is_none")]
-    log_file: Option<PathBuf>,
+    /// This should mainly be used to mount the right routs and
```
Comment (janpetschexain) [inline discovery_engine_core/web-api2/src/server.rs:47]: :art: typo
```suggestion
    /// This should mainly be used to mount the right routes and
```

File: discovery_engine_core/web-api2/src/server.rs [hunk 4]
```diff
@@ -26,38 +28,33 @@ use actix_web::{
...
+    /// application specific middleware.
+    fn configure_service(config: &mut ServiceConfig);
 
-    /// Use given configuration file.
-    #[arg(short, long)]
-    #[serde(skip_serializing_if = "Option::is_none")]
-    config: Option<PathBuf>,
+    /// Create a application specific extension to app state.
```
Comment (janpetschexain) [inline discovery_engine_core/web-api2/src/server.rs:51]: :art: typo
```suggestion
    /// Create an application specific extension to app state.
```

**Review by janpetschexain — COMMENTED** [review]
looks good, but please rebase your pr after your base branch has been merged into main, otherwise it's really annoying to review and comment on a pr, especially if there are further unrelated conflicts with main.

**Review by janpetschexain — APPROVED** [review]
👍

**Comment by rustonaut** [comment]
bors merge

**Pull request closed without merging**